tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -79,6 +84,53 @@ def test_slurm_gpu(region, pcluster_config_reader, clusters_factory):
     assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
 
 
+def _test_mpi_job_termination(remote_command_executor, test_datadir):
+    """
+    Test canceling mpirun job will not leave stray processes.
+
+    IntelMPI is known to leave stray processes after job termination if slurm process tracking is not setup correctly,
+    i.e. using ProctrackType=proctrack/pgid
+    Test IntelMPI script to make sure no stray processes after the job is cancelled
+    This bug cannot be reproduced using OpenMPI
+    Test should run on all OSs except for centos6, where IntelMPI is not available
+    """
+    logging.info("Testing no stray process left behind after mpirun job is terminated")
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    # Submit mpi_job, which runs Intel MPI benchmarks with intelmpi
+    # Leaving 1 vcpu on each node idle so that the process check job can run while mpi_job is running
+    result = slurm_commands.submit_script(str(test_datadir / "mpi_job.sh"))
+    job_id = slurm_commands.assert_job_submitted(result.stdout)
+
+    # Check that mpi processes are started
+    _assert_job_state(remote_command_executor, job_id, job_state="RUNNING")
+    _check_mpi_process(remote_command_executor, slurm_commands, test_datadir, num_nodes=2, after_completion=False)
+    slurm_commands.cancel_job(job_id)
+
+    # Make sure mpirun job is cancelled
+    _assert_job_state(remote_command_executor, job_id, job_state="CANCELLED")
+
+    # Check that mpi processes are terminated
+    _check_mpi_process(remote_command_executor, slurm_commands, test_datadir, num_nodes=2, after_completion=True)
+
+    # Wait for cluster to scale compute back down to 0
+    _assert_no_nodes_in_scheduler(slurm_commands)
```

Comment:
if this is a precondition for the next test I would move it there, so no matter how we reorder them they should always work.